### PR TITLE
Fix model renames

### PR DESCRIPTION
**Pull request by mrjackv**

### CODE REVIEW

File: include/revng/EarlyFunctionAnalysis/FunctionMetadataCache.h [hunk 1]
```diff
@@ -28,8 +28,7 @@ extractFunctionMetadata(llvm::MDNode *MD) {
 
   StringRef YAMLString = cast<MDString>(Op)->getString();
   auto MaybeParsed = TupleTree<efa::FunctionMetadata>::deserialize(YAMLString);
-  revng_assert(MaybeParsed);
-  MaybeParsed->verify();
+  revng_assert(MaybeParsed && MaybeParsed->verify());
```
Comment (aleclearmind) [inline include/revng/EarlyFunctionAnalysis/FunctionMetadataCache.h:31]:
```suggestion
  revng_assert(MaybeParsed and MaybeParsed->verify());
```

File: lib/PipelineC/PipelineC.cpp [hunk 2]
```diff
@@ -725,41 +722,42 @@ inline bool rp_manager_apply_diff_impl(rp_manager *manager,
...
+  auto ApplyError = GlobalClone->applyDiff(Diff);
+  if (ApplyError) {
+    llvmErrorToRpError(std::move(ApplyError), Error);
     return false;
   }
 
-  if (auto Error = GlobalClone->verify(); Error) {
-    **error = rp_simple_error(std::string("could not verify ") + global_name,
-                              "");
+  if (!GlobalClone->verify()) {
```
Comment (aleclearmind) [inline lib/PipelineC/PipelineC.cpp:750]: Everywhere. More readable.

```suggestion
  if (not GlobalClone->verify()) {
```

File: lib/PipelineC/PipelineC.cpp [hunk 3]
```diff
@@ -915,31 +913,31 @@ bool rp_diff_map_is_empty(rp_diff_map *map) {
...
     return nullptr;
 
-  return &std::get<rp_simple_error>(**error);
+  return &std::get<rp_simple_error>(*error);
 }
 
 rp_document_error *rp_error_get_document_error(rp_error *error) {
-  if (error->get() == nullptr
-      || not std::holds_alternative<rp_document_error>(**error))
+  if (error->index() == 0
```
Comment (aleclearmind) [inline lib/PipelineC/PipelineC.cpp:936]: Use [`std::holds_alternative`](https://en.cppreference.com/w/cpp/utility/variant/holds_alternative).

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: lib/PipelineC/PipelineC.cpp (hunk 2 above)
Comment (mrjackv) [reply lib/PipelineC/PipelineC.cpp:750]: Done

File: lib/PipelineC/PipelineC.cpp (hunk 3 above)
Comment (mrjackv) [reply lib/PipelineC/PipelineC.cpp:936]: Re-checked, the test was redundant. Dropped

**Comment by aleclearmind** [comment]
Merged in 2fdfaef. Thanks!

**Pull request closed without merging**